api/types.proto

```diff
@@ -459,12 +459,26 @@ enum TaskState {
 	// Remove this message when the states are deemed perfect.
 }
 
+// State of Networks attached to a container
+message ContainerNetworkStatus {
+	// Network is the Network ID
+	string network = 1;
+
+	// Interface is the device name as seen within the container
+	string interface = 2;
+
+	// Address is the IP address in the form ipv4:XXX or ipv6:YYY
+	string address = 3;
```

Comment:
Related to this I don't really like what I've done wrt error reporting at https://github.com/ijc/swarmkit/blob/9c619b516e62f103df460dfc8ffd9652634733ac/agent/exec/containerd/controller.go#L66, essentially it just uses an "error:" prefix to distinguish.

I think maybe it would be better to have a separate error field, probably as a `oneof`.